src/main/java/net/masterthought/cucumber/generators/EscapeHtmlReference.java

```diff
@@ -2,16 +2,27 @@
 
 import org.apache.commons.lang.StringEscapeUtils;
 import org.apache.velocity.app.event.ReferenceInsertionEventHandler;
+import org.owasp.html.HtmlPolicyBuilder;
+import org.owasp.html.PolicyFactory;
 
 /**
  * Escapes all html and xml that was provided in a reference before inserting it into a template.
+ *
+ * References that start with $_sanitize_ will be sanitized to allow urls.
  */
 final class EscapeHtmlReference implements ReferenceInsertionEventHandler {
 
+    private static final PolicyFactory LINKS = new HtmlPolicyBuilder()
+            .allowStandardUrlProtocols().allowElements("a").allowAttributes("href")
```

Comment:
Can't allow everything. There are different levels of Jenkins administrator. 

I'm thinking it would be neater if we created an exhaustive list of data that can be used as classification and explicitly allow those.

Not that it should be very long list
 * Text
 * Links
 * Images

Thats about it I think.